browser/ui/toolbar/brave_app_menu_model_browsertest.cc

```diff
@@ -5,102 +5,113 @@
-  // -1 means |model| doesn't have passed command id.
-  EXPECT_NE(-1, normal_model.GetIndexOfCommandId(IDC_SHOW_BRAVE_REWARDS));
-  EXPECT_NE(-1, normal_model.GetIndexOfCommandId(IDC_SHOW_BRAVE_SYNC));
-#if BUILDFLAG(BRAVE_WALLET_ENABLED)
-  EXPECT_NE(-1, normal_model.GetIndexOfCommandId(IDC_SHOW_BRAVE_WALLET));
-#else
-  EXPECT_EQ(-1, normal_model.GetIndexOfCommandId(IDC_SHOW_BRAVE_WALLET));
-#endif
+void CheckCommandsAreInOrderInMenuModel(
+    Browser* browser,
+    const std::vector<int>& commands_in_order) {
+  auto* browser_view = BrowserView::GetBrowserViewForBrowser(browser);
+  BraveAppMenuModel model(browser_view->toolbar(), browser);
+  model.Init();
+
+  std::vector<int> commands_index;
+  for (int id : commands_in_order) {
+    int index = model.GetIndexOfCommandId(id);
+    EXPECT_NE(-1, index);
+    commands_index.push_back(index);
+  }
+  EXPECT_TRUE(std::is_sorted(std::begin(commands_index),
+                             std::end(commands_index)));
+}
 
-  auto* command_controller = browser()->command_controller();
+// Test brave menu order test.
+// Brave menu is inserted based on corresponding commands enable status.
+// So, this doesn't test for each profiles(normal, private, tor and guest).
+// Instead, BraveBrowserCommandControllerTest will do that.
+IN_PROC_BROWSER_TEST_F(BraveAppMenuBrowserTest, MenuOrderTest) {
+  std::vector<int> commands_in_order_for_normal_profile = {
+    IDC_NEW_TAB,
+    IDC_NEW_WINDOW,
+    IDC_NEW_INCOGNITO_WINDOW,
 #if BUILDFLAG(ENABLE_TOR)
-  EXPECT_NE(
-      -1, normal_model.GetIndexOfCommandId(IDC_NEW_OFFTHERECORD_WINDOW_TOR));
-  // Check new tor browser command is only enabled for normal window.
-  EXPECT_FALSE(
-      command_controller->IsCommandEnabled(IDC_NEW_TOR_CONNECTION_FOR_SITE));
-  EXPECT_TRUE(
-      command_controller->IsCommandEnabled(IDC_NEW_OFFTHERECORD_WINDOW_TOR));
+    IDC_NEW_OFFTHERECORD_WINDOW_TOR,
 #endif
-
 #if BUILDFLAG(BRAVE_REWARDS_ENABLED)
-  EXPECT_TRUE(command_controller->IsCommandEnabled(IDC_SHOW_BRAVE_REWARDS));
-#else
-  EXPECT_FALSE(command_controller->IsCommandEnabled(IDC_SHOW_BRAVE_REWARDS));
+    IDC_SHOW_BRAVE_REWARDS,
 #endif
-
-#if BUILDFLAG(ENABLE_BRAVE_SYNC)
-  EXPECT_TRUE(command_controller->IsCommandEnabled(IDC_SHOW_BRAVE_SYNC));
-#else
-  EXPECT_FALSE(command_controller->IsCommandEnabled(IDC_SHOW_BRAVE_SYNC));
-#endif
-
+    IDC_RECENT_TABS_MENU,
+    IDC_BOOKMARKS_MENU,
+    IDC_SHOW_DOWNLOADS,
 #if BUILDFLAG(BRAVE_WALLET_ENABLED)
-  EXPECT_TRUE(command_controller->IsCommandEnabled(IDC_SHOW_BRAVE_WALLET));
-#else
-  EXPECT_FALSE(command_controller->IsCommandEnabled(IDC_SHOW_BRAVE_WALLET));
+    IDC_SHOW_BRAVE_WALLET,
 #endif
+    IDC_MANAGE_EXTENSIONS,
+#if BUILDFLAG(ENABLE_BRAVE_SYNC)
+    IDC_SHOW_BRAVE_SYNC,
+#endif
+    IDC_SHOW_BRAVE_ADBLOCK,
+    IDC_ADD_NEW_PROFILE,
+    IDC_OPEN_GUEST_PROFILE,
+    IDC_SHOW_BRAVE_WEBCOMPAT_REPORTER
+  };
+  std::vector<int> commands_disabled_for_normal_profile = {
```

Comment:
shouldn't we check index of these commands are all -1?

Reply:
Oh, thanks for catching!